src/GPE.jl

```diff
@@ -150,20 +173,33 @@ end
 
 Update the covariance matrix and its Cholesky decomposition of Gaussian process `gp`.
 """
-function update_cK!(gp::GPE)
+function update_cK!(gp::GPE{X,Y,M,K,P,D}) where {X,Y,M,K,P,D}
     old_cK = gp.cK
-    Σbuffer = old_cK.mat
+    if P <: ElasticPDMat
```

Comment:
Again, let's use multiple dispatch here. We could put all the ElasticPDMat methods in their own module to avoid cluttering `GPE.jl`, as many users won't be interested in this usecase.